### PR TITLE
feat: Implement gas relay mode and inclusion data for data attestation

### PR DESCRIPTION
**Pull request by QEDK**
## What ❔

This PR adds gas relay API support for gasless submission to the Avail network, it also provides the attestation implementation necessary for data attestation.

## Why ❔

Gas relay API support is required for Avail partners that choose to pay in a different token. Data attestation ensures that arbitrary tx data cannot be used for rollup finality and that no data withholding attack can occur.

## Checklist

- [X] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [X] Documentation comments have been added / updated.
- [X] Code has been formatted via `zk fmt` and `zk lint`.

### CODE REVIEW

File: core/node/da_clients/src/avail/client.rs [hunk 1]
```diff
@@ -35,37 +108,165 @@ impl DataAvailabilityClient for AvailClient {
...
+            }
+        }
+        let attestation_data: MerkleProofInput = MerkleProofInput {
+            dataRootProof: bridge_api_data.data_root_proof.unwrap(),
+            leafProof: bridge_api_data.leaf_proof.unwrap(),
+            rangeHash: bridge_api_data.range_hash.unwrap(),
+            dataRootIndex: bridge_api_data.data_root_index.unwrap(),
+            blobRoot: bridge_api_data.blob_root.unwrap(),
+            bridgeRoot: bridge_api_data.bridge_root.unwrap(),
+            leaf: bridge_api_data.leaf.unwrap(),
```
Comment (dimazhornyk) [inline core/node/da_clients/src/avail/client.rs:264]: is this a simple `keccak(data)` same as in L2DAValidator contract?

**Review by dimazhornyk — COMMENTED** [review]
left some high level comments to note when changing PR destination branch

File: core/lib/protobuf_config/src/da_client.rs [hunk 2]
```diff
@@ -18,16 +18,53 @@ impl ProtoRepr for proto::DataAvailabilityClient {
 
         let client = match config {
             proto::data_availability_client::Config::Avail(conf) => Avail(AvailConfig {
-                api_node_url: required(&conf.api_node_url)
-                    .context("api_node_url")?
-                    .clone(),
+                api_node_url: if conf.gas_relay_mode.unwrap_or(false) {
```
[the remark this answers is not in the document]
Comment (dimazhornyk) [reply core/lib/protobuf_config/src/da_client.rs:21]: or maybe even consider doing the gas relay a separate client alongside AvailClient, as both the config management and the actual implementation logic is different

File: Cargo.toml [hunk 3]
```diff
@@ -102,6 +102,7 @@ categories = ["cryptography"]
 
 [workspace.dependencies]
 # "External" dependencies
+alloy = { version = "0.1", features = ["serde", "sol-types"] }
```
[the remark this answers is not in the document]
Comment (QEDK) [reply Cargo.toml:105]: I had a quick look but unfortunately it wasn't immediately clear how this works using the more native approaches, hence implemented using `alloy` which is quite readable, can you link me to the exact implementation using `Tokenize` and `Tokenizable`?

File: core/node/da_clients/src/avail/client.rs (hunk 1 above)
Comment (QEDK) [reply core/node/da_clients/src/avail/client.rs:264]: That is correct.

File: Cargo.toml (hunk 3 above)
[the remark this answers is not in the document]
Comment (QEDK) [reply Cargo.toml:105]: This is done in the latest PR, closing.

File: core/lib/protobuf_config/src/da_client.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (QEDK) [reply core/lib/protobuf_config/src/da_client.rs:21]: This is done in the latest PR, closing.

File: core/lib/protobuf_config/src/proto/config/da_client.proto [hunk 4]
```diff
@@ -11,6 +11,9 @@ message AvailConfig {
   optional uint32 app_id = 4;
   optional uint64 timeout = 5;
   optional uint64 max_retries = 6;
+  optional bool gas_relay_mode = 7;
+  optional string gas_relay_api_url = 8;
+  optional string gas_relay_api_key = 9;
```
[the remark this answers is not in the document]
Comment (QEDK) [reply core/lib/protobuf_config/src/proto/config/da_client.proto:16]: This is done in the latest PR, closing.

File: core/lib/config/src/configs/da_client/avail.rs [hunk 5]
```diff
@@ -2,10 +2,13 @@ use serde::Deserialize;
...
     pub bridge_api_url: String,
-    pub seed: String,
-    pub app_id: u32,
+    pub seed: Option<String>,
+    pub app_id: Option<u32>,
     pub timeout: usize,
     pub max_retries: usize,
+    pub gas_relay_mode: bool,
+    pub gas_relay_api_url: Option<String>,
+    pub gas_relay_api_key: Option<String>,
```
[the remark this answers is not in the document]
Comment (QEDK) [reply core/lib/config/src/configs/da_client/avail.rs:13]: This is done in the latest PR, closing.

File: Cargo.lock [hunk 6]
```diff
@@ -15,18 +15,18 @@ dependencies = [
 
 [[package]]
 name = "addr2line"
-version = "0.21.0"
```
[the remark this answers is not in the document]
Comment (QEDK) [reply Cargo.lock:18]: This is done in the latest PR, closing.

**Comment by QEDK** [comment]
Superseded by #3070 - closing.

**Pull request closed without merging**